csrc/sign.cpp

```diff
@@ -66,7 +66,7 @@ bool initializeContext(raii_env& env,
     EVP_PKEY_up_ref(pKey);
     ctx->setKey(pKey);
 
-    if (md != nullptr) {
+    if (md != nullptr || EVP_PKEY_id(pKey) == EVP_PKEY_ED25519) {
```

Comment:
`md` is implicitly null, but we need to follow that path for Ed25519 signatures, since we are using the EVP_DigestSign method that relies on DigestCtx being set (as opposed to the KeyCtx being used, as in L84).